packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -170,8 +170,8 @@ const VirtualScrollerPinnedRows = styled('div', {
   const boxShadowColor = getBoxShadowColor(theme);
   return {
     position: 'sticky',
-    // should be above the detail panel
-    zIndex: 3,
+    // should be above the no rows overlay
+    zIndex: 5,
```

Comment:
I like @yaredtsy's suggestion of adding the message over the unpinned content instead of over the whole content, but I'm not strongly opposed to this.

Reply:
Yes, showing the overlay only over the unpinned content + making the text `No unpinned rows` would be a better XP
Or better, 
```js
const message = pinnedRows.length > 0 ? 'No unpinned rows' : 'No rows'
```

Reply:
This combination is probably the best direction. I'll update the PR accordingly.